src/NUglify/JavaScript/Visitors/OutputVisitor.cs

```diff
@@ -3985,15 +3985,22 @@ void OutputFunctionArgsAndBody(FunctionObject node)
                     Unindent();
                     if (wrapInParens)
                     {
-                        OutputPossibleLineBreak(')');
+                        if (node.FunctionType == FunctionType.ArrowFunction) 
+                        {
+                            Output(')');
+                        }
+                        else 
+                        {
+                            OutputPossibleLineBreak(')');
+                        }

```

Comment:
Are you sure this is needed?

Reply:
Very sure: If you generate the code without this, you will get a compile-error for the generated code.
But I don't like the lack of elegance in this change either. I thought about creating a new overload for Output: Output(char, bool allowLineBreak). But I did not want to be too intrusive and had no better idea.

In fact, I was unable to reach the code in line 4003 below with the unit-tests. But I left the change there anyway, since it is pretty obvious that a line-break would lead to invalid code.